code/game/objects/items/shuttle_creator.dm

```diff
@@ -0,0 +1,299 @@
+	port.initiate_docking(stationary_port)
+
+	port.mode = SHUTTLE_IDLE
+	port.timer = 0
+
+	port.register()
+	GLOB.custom_shuttle_count ++
+	message_admins("[ADMIN_LOOKUPFLW(user)] created a new shuttle with a [src] at [ADMIN_VERBOSEJMP(user)] ([GLOB.custom_shuttle_count] custom shuttles, limit is [CUSTOM_SHUTTLE_LIMIT])")
+	log_game("[key_name(user)] created a new shuttle with a [src] at [AREACOORD(user)] ([GLOB.custom_shuttle_count] custom shuttles, limit is [CUSTOM_SHUTTLE_LIMIT])")
+	return TRUE
+
+/obj/item/shuttle_creator/proc/create_shuttle_area(mob/user)
+	if(!loggedTurfs)
+		return FALSE
+	//Create the new area
+	var/area/shuttle/custom/powered/newS
+	var/area/oldA = loggedOldArea
+	var/str = stripped_input(user, "Shuttle Name:", "Blueprint Editing", "", MAX_NAME_LEN)
+	if(!str || !length(str))
+		return FALSE
+	if(length(str) > 50)
+		to_chat(user, "<span class='warning'>The provided ship name is too long, blares the [src]</span>")
+		return FALSE
+	if(OOC_FILTER_CHECK(str))
+		to_chat(user, "<span class='warning'>Nanotrasen prohibited words are in use in this shuttle name, blars the [src] in a slightly offended tone.</span>")
+		return FALSE
+	newS = new /area/shuttle/custom/powered()
+	newS.setup(str)
+	newS.set_dynamic_lighting()
+	//Shuttles always have gravity
+	newS.has_gravity = TRUE
+	newS.requires_power = TRUE
+	//Record the area for use when creating the docking port
+	recorded_shuttle_area = newS
+
+	for(var/i in 1 to loggedTurfs.len)
+		var/turf/turf_holder = loggedTurfs[i]
+		var/area/old_area = turf_holder.loc
+		newS.contents += turf_holder
+		turf_holder.change_area(old_area, newS)
+
+	newS.reg_in_areas_in_z()
+
+	var/list/firedoors = oldA.firedoors
+	for(var/door in firedoors)
+		var/obj/machinery/door/firedoor/FD = door
+		FD.CalculateAffectingAreas()
+	return TRUE
+
+/obj/item/shuttle_creator/proc/check_current_area(mob/user)
+	var/static/area_or_turf_fail_types = typecacheof(list(
+		/turf/open/space,
+		/area/shuttle
+		))
+	//Check to see if the user can make a new area to prevent spamming
+	if(user)
+		if(user.create_area_cooldown >= world.time)
+			to_chat(user, "<span class='warning'>Smoke vents from the [src], maybe you should let it cooldown before using it again.</span>")
+			return
+		user.create_area_cooldown = world.time + 10
+	//Detect the turfs connected in the curerrent enclosed area
+	var/list/turfs = detect_room(get_turf(user), area_or_turf_fail_types)
+	if(!turfs)
+		to_chat(user, "<span class='warning'>Shuttles must be created in an airtight space, ensure that the shuttle is airtight, including corners.</span>")
+		return
+	if(turfs.len > SHUTTLE_CREATOR_MAX_SIZE)
+		to_chat(user, "<span class='warning'>The [src]'s internal cooling system wizzes violently and a message appears on the screen, \"Caution, this device can only handle the creation of shuttles up to [SHUTTLE_CREATOR_MAX_SIZE]. Please reduce your shuttle by [turfs.len-SHUTTLE_CREATOR_MAX_SIZE]. Sorry for the inconvinience\"</span>")
```

Comment:
You should have it say turfs after the amount of turfs, so that it says "can only create shuttles up to [x] turfs in size," instead of "can only create shuttles up to [x]."